README.md

```diff
@@ -1,5 +1,131 @@
 # snips-utils-rs
 
+This crate provides a framework to convert idiomatic Rust structs to C-like structs that can pass through an FFI boundary, and conversely.
+
+This framework is made up of two **conversion traits**, **`CReprOf`** and **`AsRust`**.
+They ensure that the developper uses best practices when performing the conversion in both directions (ownership-wise).
+
+The crate also provides a collection of useful utility functions to perform conversions of types.
+It goes hand in hand with the `ffi-utils-derive` as it provides an **automatic derivation** of the `CReprOf` and `AsRust` trait.
+
+## Usage
+
+We want to be able to convert a **`Pizza`** Rust struct that has an idiomatic representation to a **`CPizza`** Rust struct that
+has a C-compatible representation in memory.
+
+We start by definining the fields of the Pizza struct :
+
+```
+pub struct Pizza {
+    pub name: String,
+    pub toppings: Vec<Topping>,
+    pub base: Option<Sauce>,
+    pub weight: f32,
+}
+```
+
+We then create the C-like struct by mapping idiomatic Rust types to C compatible types :
+
+```
+#[repr(C)]
+pub struct CPizza {
+    pub name: *const libc::c_char,
+    pub toppings: *const CArray<CTopping>,
+    pub base: *const CSauce,
+    pub weight: libc::c_float,
```

Comment:
If using the libc wrapper, it doesn't work no? We may want to indicate to use the primitive notation (`f32`) rather than this. From what I've experienced, this is quite tricky. The compiler doesn't point out  where the issue is. 

for example:
```
error: expected identifier, found `::`
  --> sve/sve-trampoline/sve-trampoline-client-ffi/src/slu.rs:24:10
   |
24 | #[derive(CReprOf, AsRust, CDrop)]
   |          ^^^^^^^ expected identifier

error: proc-macro derive produced unparseable tokens
```